src/util/file_io.c

```diff
@@ -17,20 +17,21 @@
  *****************************************************************************/
  #include <stdint.h>
  #include <stdio.h>
+ #include <unistd.h>
  #include <xtt/util/file_io.h>
  #include <xtt/util/util_errors.h>
 
- int xtt_save_to_file(unsigned char *buffer, size_t bytes_to_write, const char *filename)
+ int xtt_save_to_file(unsigned char *buffer, size_t bytes_to_write, const char *filename, mode_t permission)
  {
-     FILE *file_ptr = fopen(filename, "wb");
-     int ret = 0;
-     int close_ret = 0;
 
-     if (NULL == file_ptr){
-         return SAVE_TO_FILE_ERROR;
+     int fd = open(filename, O_WRONLY | O_CREAT, permission);
+     if (fd == -1){
+        return SAVE_TO_FILE_ERROR;
      }
 
-     size_t bytes_written = fwrite(buffer, 1, bytes_to_write, file_ptr);
+     size_t bytes_written = write(fd, buffer, bytes_to_write);
+     int ret = 0;
+     int close_ret = 0;
```

Comment:
Since we never have to use `close_ret`, other than checking it, let's get rid of it and just do
```
if (-1 == close(fd)) {
...
```